storm-server/src/main/java/org/apache/storm/daemon/nimbus/Nimbus.java

```diff
@@ -4652,4 +4745,192 @@ public IScheduler getForcedScheduler() {
 
     }
 
+    //enum NotPorted {
+    //    //Declared in StormConf. I don't see the value in reporting so.
+    //    SUPERVISOR_TOTAL_RESOURCE,
+    //    //May be able to aggregate based on status;
+    //    TOPOLOGY_STATUS,
+    //    TOPOLOGY_SCHED_STATUS,
+    //    //May be aggregated, as well as other distinct values
+    //    NUM_DISTINCT_NIMBUS_VERSION;
+    //}
+
+    private class ClusterSummaryMetricSet implements MetricSet, Runnable {
```

Comment:
Can this class be moved to another file? Nimbus is already huge.

Reply:
ClusterSummaryMetricSet rely on `isLeader` from Nimbus.

Reply:
Then `isLeader` can be made public or protected, and Nimbus can be injected into this class via a constructor.

Reply:
Additionally, we can't call `getClusterInfoImpl` directly, so we have to use the public thrift API `getClusterInfo`. This will disrupt the accuracy of its meter "nimbus:num-getClusterInfo-calls", because I think it's for tracking public API invocation only.

Reply:
That can be solved by making `getClusterInfoImpl` package private, but I see your point. Maybe we just leave it here for now, and I'll see if we can do something with it when I look at the non-static registry branch. It's not like it hurts anything to have it here, I'd just rather have it in another file if possible.

Reply:
Yes, I agree. The current implementation isn't ideal.